waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -189,17 +187,46 @@
         (is (= expected-namespace pod-namespace))))))
 
 (deftest ^:parallel ^:integration-fast test-pod-namespace
+  "Expected behavior for services with namespaces:
+   Run-As-User    Namespace   Validation
+   Missing        Missing     OK
```

Comment:
Added the matrix from your comment into the docstring here. All 12 cases are explicitly tested below.